internal/server/receiver_handler_test.go

```diff
@@ -639,6 +639,10 @@ func Test_handlePayload(t *testing.T) {
 			if tt.receiver != nil {
 				builder.WithObjects(tt.receiver)
 			}
+			builder.WithIndex(&apiv1.Receiver{}, WebhookPathIndexKey, func(o client.Object) []string {
+				receiver := o.(*apiv1.Receiver)
+				return []string{receiver.Status.WebhookPath}
```

Comment:
yes, i thought about this too, but then decided against it, because we'd need to define the index func in the `internal/server` package but then no actual code in the package would be using it (only a single test), whereas the main place of its use would be in `internal/controller`, so i find that a bit weird, semantically speaking.
but if you think its fine to have such a thing, then i don't really mind moving it.